cmake/rename.cmake

```diff
@@ -0,0 +1,25 @@
+# Only for CACHE variable (option)
+macro(gko_rename_cache deprecated actual type)
+    if(DEFINED ${deprecated})
+        if(DEFINED ${actual})
+            message("actual ${actual} and deprecated ${deprecated}")
+            if("${${actual}}" STREQUAL "${${deprecated}}")
+                # They are the same, so only throw warning
+                message(WARNING "${deprecated} was deprecated, please only use ${actual} instead.")
+            else()
+                # They are different
+                set(${deprecated}_copy ${${deprecated}})
+                unset(${deprecated} CACHE)
+                message(FATAL_ERROR "Both ${deprecated} and ${actual} were specified, please use ${actual} instead.  "
```

Comment:
This error message would vanish after rerunning CMake, is that intended? Maybe people might miss it under the large amount of output, and just rerun CMake, silently losing their configuration option.

Reply:
I currently do not unset the deprecated variable.
It will keep showing the warning or error when the deprecated is there